providers/dns/mijnhost/mijnhost.go

```diff
@@ -0,0 +1,212 @@
+		SequenceInterval:   env.GetOrDefaultSecond(EnvSequenceInterval, 5*time.Second),
+		HTTPClient: &http.Client{
+			Timeout: env.GetOrDefaultSecond(EnvHTTPTimeout, 30*time.Second),
+		},
+	}
+}
+
+// DNSProvider implements the challenge.Provider interface.
+type DNSProvider struct {
+	config *Config
+	client *internal.Client
+}
+
+// NewDNSProvider returns a DNSProvider instance configured for mijn.host DNS.
+// MIJN_API_KEY must be passed in the environment variables.
+func NewDNSProvider() (*DNSProvider, error) {
+	values, err := env.Get(EnvAPIKey)
+	if err != nil {
+		return nil, fmt.Errorf("mijnhost: %w", err)
+	}
+
+	config := NewDefaultConfig()
+	config.APIKey = values[EnvAPIKey]
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for mijn.host DNS.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		return nil, errors.New("mijnhost: the configuration of the DNS provider is nil")
+	}
+
+	if config.APIKey == "" {
+		return nil, errors.New("mijnhost: APIKey is missing")
+	}
+
+	client := internal.NewClient(config.APIKey)
+
+	return &DNSProvider{
+		config: config,
+		client: client,
+	}, nil
+}
+
+// Timeout returns the timeout and interval to use when checking for DNS propagation.
+// Adjusting here to cope with spikes in propagation times.
+func (d *DNSProvider) Timeout() (timeout, interval time.Duration) {
+	return d.config.PropagationTimeout, d.config.PollingInterval
+}
+
+// Sequential All DNS challenges for this provider will be resolved sequentially.
+// Returns the interval between each iteration.
+func (d *DNSProvider) Sequential() time.Duration {
+	return d.config.SequenceInterval
+}
+
+// Present creates a TXT record to fulfill the dns-01 challenge.
+func (d *DNSProvider) Present(domain, token, keyAuth string) error {
+	info := dns01.GetChallengeInfo(domain, keyAuth)
+
+	domains, err := d.client.ListDomains(context.Background())
+	if err != nil {
+		return fmt.Errorf("mijnhost: list domains: %w", err)
+	}
+
+	dom, err := findDomain(domains, domain)
+	if err != nil {
+		return fmt.Errorf("mijnhost: find domain: %w", err)
+	}
+
+	records, err := d.client.GetRecords(context.Background(), dom.Domain)
+	if err != nil {
+		return fmt.Errorf("mijnhost: get records: %w", err)
+	}
+
+	subDomain, err := dns01.ExtractSubDomain(info.EffectiveFQDN, dom.Domain)
+	if err != nil {
+		return fmt.Errorf("mijnhost: %w", err)
+	}
+
+	record := internal.Record{
+		Type:  "TXT",
+		Name:  subDomain,
+		Value: info.Value,
+		TTL:   d.config.TTL,
+	}
+
+	// mijn.host doesn't support multiple values for a domain,
+	// so we removed existing record for the subdomain.
+	cleanedRecords := filterRecords(records, func(record internal.Record) bool {
+		// TODO(ldez) I don't known if the records returned by the endpoint have the same content as name:
+		// - I use subdomain to create the records
+		// - In the API example the name is the full domain
```

Comment:
@shubhamku044, would you mind checking whether this PR works with a subdomain? Something like

```
MIJNHOST_API_KEY=yyyy \
./dist/lego -m your@email.com \
  --dns mijnhost \
  -d www.testdomein12345.nl \
  -s https://acme-staging-v02.api.letsencrypt.org/directory \
  run
```

should to the trick.

Reply:
No, because it's a prevention during the creation if there is an already existing ACME TXT record not related to lego.
I will just check both: the domain and subdomain.